pkg/cloudprovider/aws/instance.go

```diff
@@ -36,11 +40,30 @@ import (
 	"github.com/awslabs/karpenter/pkg/utils/options"
 )
 
+const (
+	iceCacheODTTL           = 15 * time.Second
+	iceCacheSpotTTL         = 45 * time.Second
+	iceCacheCleanupInterval = 5 * time.Minute
+)
+
 type InstanceProvider struct {
 	ec2api                 ec2iface.EC2API
 	instanceTypeProvider   *InstanceTypeProvider
 	subnetProvider         *SubnetProvider
 	launchTemplateProvider *LaunchTemplateProvider
+	iceCache               *cache.Cache
+}
+
+func NewInstanceProvider(ec2api ec2iface.EC2API, instanceTypeProvider *InstanceTypeProvider, subnetProvider *SubnetProvider, sess *session.Session,
+	clientSet *kubernetes.Clientset) *InstanceProvider {
+	return &InstanceProvider{ec2api, instanceTypeProvider, subnetProvider,
+		NewLaunchTemplateProvider(
+			ec2api,
+			NewAMIProvider(ssm.New(sess), clientSet),
+			NewSecurityGroupProvider(ec2api),
+		),
+		cache.New(iceCacheSpotTTL, iceCacheCleanupInterval),
```

Comment:
Seems a bit odd to me to use the spot TTL since it happens to be lower. WDYT about setting to -1 to disable eviction? It's a finite set, so I'm not concerned about cache size.